### PR TITLE
Feature EPL-683: User context in request scope

### PR DESCRIPTION
**Pull request by sebastianbarrozo**

### CODE REVIEW

File: libs/com.etendorx.utils.auth/src/main/java/com/etendorx/utils/auth/key/context/FilterContext.java [hunk 1]
```diff
@@ -17,41 +17,60 @@
...
     userContext.setClientId((String) tokenValuesMap.get(JwtKeyUtils.CLIENT_ID_CLAIM));
     userContext.setOrganizationId((String) tokenValuesMap.get(JwtKeyUtils.ORG_ID));
     userContext.setRoleId((String) tokenValuesMap.get(JwtKeyUtils.ROLE_ID));
     userContext.setSearchKey((String) tokenValuesMap.get(JwtKeyUtils.SERVICE_SEARCH_KEY));
     userContext.setServiceId((String) tokenValuesMap.get(JwtKeyUtils.SERVICE_ID));
-    userContext.setActive(request.getParameter("active"));
-    userContext.setRestMethod(request.getMethod());
-    AppContext.setCurrentUser(userContext);
+    String active = Optional.ofNullable(activeParam).orElse("true");
+    if (!active.equals("true") && !active.equals("false")) {
```
Comment (juabaez) [inline libs/com.etendorx.utils.auth/src/main/java/com/etendorx/utils/auth/key/context/FilterContext.java:69]: it's possible to use StringUtils ?

File: libs/com.etendorx.utils.auth/src/main/java/com/etendorx/utils/auth/key/context/FilterContext.java [hunk 2]
```diff
@@ -17,41 +17,60 @@
...
+      throw new IllegalArgumentException("Invalid value for 'active' parameter: " + active);
+    }
+    userContext.setActive("true".equals(active));
```
Comment (juabaez) [inline libs/com.etendorx.utils.auth/src/main/java/com/etendorx/utils/auth/key/context/FilterContext.java:72]: it's possible to use StringUtils ?

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java [hunk 3]
```diff
@@ -43,11 +47,18 @@ public static String addFilters(String restMethod, String sql, String userId, St
...
+        boolean containsWhere, String tableAlias) {
+        List<String> conditions = new ArrayList<>();
+        if (isActive) {
+            conditions.add(tableAlias + ".isactive = 'Y'");
+        }
+        conditions.add(tableAlias + ".ad_client_id = '" + clientId + "'");
+        conditions.add("(" + tableAlias + ".ad_org_id =  '" + orgId + "' OR " +
+            "((etrx_is_org_in_org_tree(" + tableAlias + ".ad_org_id, '" + orgId + "', '1')) = 1))");
+        String whereClause = String.join(AND, conditions);
+        return sql.replace((containsWhere ? WHERE : LIMIT),
```
Comment (juabaez) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:60]: here we need to check witch database is in use and use limit or rownum right ?

**Review by juabaez — CHANGES_REQUESTED** [review]

**Review by juabaez — APPROVED** [review]

**Review by Matias-Bernal — APPROVED** [review]

**Review by Matias-Bernal — APPROVED** [review]

**Review by juabaez — APPROVED** [review]

**Review by Matias-Bernal — APPROVED** [review]

**Review by Matias-Bernal — APPROVED** [review]

**Comment by valeg-etendo** [comment]
@sebastianbarrozo Please, review the Sonar report

**Review by juabaez — APPROVED** [review]

**Pull request merged**